sumfields.php

```diff
@@ -1255,7 +1262,12 @@ function sumfields_gen_data(&$returnValues) {
       if(sumfields_generate_data_based_on_current_data()) {
         CRM_Core_DAO::triggerRebuild();
         $date = date('Y-m-d H:i:s');
-        $new_status = 'success:' . $date;
+        if ($process_type == 'via_cron') {
```

Comment:
$data_update_method. 

Also - I'm curious about performance here. We run our cron jobs every 10 minutes. With that setting, it seems like we would be rebuilding the summaries for every single contact every 10 minutes. That's seems a bit excessive - especially for a large data set. What if we added a setting to indicate an interval? So, we could still run the cron job every 10 minutes, but when re-scheduling here, we could set the date to be 24 hours in the future (and add code to check the date). Or.... I suppose we could add a second cron job just for this purpose.

What do you think?

Reply:
Well, from a functional perspective, on those big clients, a 10 minute cronjob is an overkill, as the cron would choke on itself: One of them has 1600 smartgroups, it takes at least 16 minutes to calculate only the smartgroups on every cron run. In reality, their cron is running every 3h. I do get the point on the comment, i'm just thinking if we're adding too much complexity..

Reply:
Seems reasonable - I agree, I don't want to kill ourselves on complexity. If you choose the via_cron method, you can simply change your cron job schedule to be different. The only possible downside is if you submit changes and specify that changes should take place via cron, you might have to wait three hours. But again, not a big deal if the trade off is avoiding complexity.